src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file preprocess_string_main.cpp
+ * @author Jeffin Sam
+ *
+ * A CLI executable to preprocess string dataset.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/math/random.hpp>
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/data/string_cleaning.hpp>
+
+PROGRAM_INFO("preprocess_string",
+    // Short description.
+    "A utility to preprocess string data. This utility can remove stopwords, "
+    "punctuation and convert to lowercase.",
+    // Long description.
+    "This utility takes a dataset and the dimension and arguments and "
+    "does the preprocessing of string dataset according to arguments given."
+    "\n\n"
+    "The dataset may be given as the file name and the output may be saved as "
+    + PRINT_PARAM_STRING("actual_dataset") + " and " +
+    PRINT_PARAM_STRING("preprocess_dataset") + " ."
+    "\n\n"
+    " Following arguments may be given " + PRINT_PARAM_STRING("lowercase") +
+    " to convert the dataset to lower case and " +
+    PRINT_PARAM_STRING("punctuation") + "for removing punctuation and "
+    + PRINT_PARAM_STRING("stopwords") + "for removing stopwords. Also the "
+    "dimension which contains the string dataset " +
+    PRINT_PARAM_STRING("dimension") + "."
+    "\n\n"
+    "So, a simple example where we want to preprocess string the dataset " +
+    PRINT_DATASET("X") + ", which is having string data in its 3 Column."
+    "\n\n" +
+    PRINT_CALL("preprocess_string", "actual_dataset", "X",
+        "preprocess_dataset", "X", "lowercase", 1, "stopwords",
+        1, "punctuation", 1, "dimension", 3) +
+    "\n\n",
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+PARAM_STRING_IN_REQ("actual_dataset", "File containing the reference dataset.",
+    "t");
+PARAM_STRING_IN_REQ("preprocess_dataset", "File containing the preprocess "
+    "dataset.", "o");
+PARAM_STRING_IN_REQ("delimeter", "delimeter used to seperate Column in files"
+    "example '\\t' for '.tsv' and ',' for '.csv'.", "d");
+
+PARAM_FLAG("lowercase", "convert to lowercase.", "l");
+PARAM_FLAG("punctuation", "Remove punctuation.", "p");
+PARAM_FLAG("stopwords", "Remove stopwords.", "s");
+PARAM_INT_IN("dimension", "Column which contains the string data.(1 by "
+    "default)", "c", 1);
```

Comment:
Maybe `PARAM_VECTOR_INT_IN` is better here?  It would allow the user to specify multiple dimensions containing string data.  It may also be worth considering an `auto` option here or something that will detect non-numeric dimensions automatically.